corehq/apps/domain/auth.py

```diff
@@ -70,7 +70,7 @@ def get_username_and_password_from_request(request):
     may be null."""
     from corehq.apps.hqwebapp.utils import decode_password
 
-    if 'HTTP_AUTHORIZATION' not in request.META:
+    if 'HTTP_AUTHORIZATION' not in request.META and 'username' not in request.GET:
```

Comment:
We have a utility to extract credentials from api key requests: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/domain/auth.py#L46-L46
You could probably use that here too rather than manually inspecting params.